### DISCUSSION TITLE
Can I get just the explicitly defined items in an interface instead of derived items?

### DISCUSSION THREAD

**Discussion by tclarke** [xesscorp/skidl#226] [Q&A] [answered]
```
myif = Interface(
  GND = Net('GND'),
  p_5v = Net('+5V'),
  A = Bus('A', 20),
  D = Bus('D', 8)
)
```
I'd like to access all the items explicitly defined. Is there a way to do this? For example:
```
for n,v in myif.explicity_items(): print(n)
```
would result in
```
GND
+5V
A
D
```
and would not include: A1, A2, A3, D1, D2, etc.

**Comment by devbisme** [comment]
An `Interface` is derived from a `dict`. You should be able to do this:
```
for n, v in myif.items(): print(n)
```

**Reply by tclarke** [reply]
I think there's a misunderstanding. If I create:
for myif above, your code would print:
GND
+5V
D
D1
D2
D3
D4
....
I only want: GND, +5V, D, and A not the derived wires in the busses.

**Reply by tclarke** [reply]
the use case is, an Interface and a module with similarly named items, I want to have a loop that automatically connects the items with the same names. If I loop everything, I'll either get warnings about duplicate connections or multiple connections for the same wires.

**Reply by devbisme** [reply, marked as answer]
I added an `unexpio` dict that stores the interface I/O without expanding any buses:
```
myif = Interface(GND=Net('GND'), p_5v=Net('+5V'), A=Bus('A',20), D=Bus('D',8))
for n,v in myif.unexpio.items(): print(v.name)

GND
+5V
A
D
```
This is in the `development` branch so you can try it with:
```
pip install git+https://github.com/devbisme/skidl@development
```
Let me know if this is acceptable.

**Reply by tclarke** [reply]
Look good! thanks

**Reply by devbisme** [reply]
In the new 2.0.0 release of SKiDL, I've implemented your idea for connecting similarly-named items between two interfaces (I think). If you have two interfaces `intfc1` and `intfc2`, entries with the same name in both will be connected using either `intfc1 += intfc2` or `intfc1.connect(intfc2)`.
